platform/testing/end-to-end/vsp-cypress-resources.md

```diff
@@ -0,0 +1,835 @@
+
+Purpose: Sets up appointments with confirmed VA & CC dates, a cancellation, and appointment request message.
+
+Reference:
+
+```javascript
+export function initAppointmentListMock() {
+  cy.server();
+  cy.login();
+  mockFeatureToggles();
+  mockSupportedSites();
+
+  cy.route({
+    method: "GET",
+    url: "/vaos/v0/appointment_requests*",
+    response: updateRequestDates(requests),
+  });
+
+  cy.route({
+    method: "GET",
+    url: /.*\/v0\/appointments.*type=va$/,
+    response: updateConfirmedVADates(confirmedVA),
+  });
+
+  cy.route({
+    method: "GET",
+    url: /.*\/v0\/appointments.*type=cc$/,
+    response: updateConfirmedCCDates(confirmedCC),
+  });
+
+  cy.route({
+    method: "GET",
+    url: "/vaos/v0/facilities/983/cancel_reasons",
+    response: cancelReasons,
+  });
+
+  cy.route({
+    method: "PUT",
+    url: "/vaos/v0/appointments/cancel",
+    response: "",
+  });
+
+  cy.route({
+    method: "GET",
+    url:
+      "/vaos/v0/appointment_requests/8a48912a6cab0202016cb4fcaa8b0038/messages",
+    response: {
+      data: [
+        {
+          id: "8a48912a6cab0202016cb4fcaa8b0038",
+          type: "messages",
+          attributes: {
+            surrogateIdentifier: {},
+            messageText: "Request 2 Message 1 Text",
+            messageDateTime: "11/11/2019 12:26:13",
+            senderId: "1012845331V153043",
+            appointmentRequestId: "8a48912a6cab0202016cb4fcaa8b0038",
+            date: "2019-11-11T12:26:13.931+0000",
+            assigningAuthority: "ICN",
+            systemId: "var",
+          },
+        },
+      ],
+    },
+  });
+}
+```
+
+#### createPastVAAppointments <a name="create-past-va-appointments"></a>
+
+Import: `import { initAppointmentListMock, createPastVAAppointments, } from './vaos-cypress-helpers'`
+
+Purpose: Creates two appointments, one three days ago and another four months ago.
+
+Reference:
+
+```javascript
+export function createPastVAAppointments() {
+  const appointments = [];
+  let appointment = getVAAppointmentMock();
+  appointment.attributes = {
+    ...appointment.attributes,
+    startDate: moment().add(-3, "days").format(),
+    clinicFriendlyName: "Three day clinic name",
+    facilityId: "983",
+    sta6aid: "983GC",
+  };
+  appointment.attributes.vdsAppointments[0].currentStatus = "CHECKED OUT";
+  appointments.push(appointment);
+
+  appointment = getVAAppointmentMock();
+  appointment.attributes = {
+    ...appointment.attributes,
+    startDate: moment().add(-4, "months").format(),
+    clinicFriendlyName: "Four month clinic name",
+    facilityId: "983",
+    sta6aid: "983GC",
+  };
+  appointment.attributes.vdsAppointments[0].currentStatus = "CHECKED OUT";
+  appointments.push(appointment);
+
+  return {
+    data: appointments,
+  };
+}
+```
+
+#### initCommunityCareMock <a name="init-community-care-mock"></a>
+
+Import: `import { initCommunityCareMock } from './vaos-cypress-helpers';`
+
+Purpose:
```

Comment:
Is there a defined purpose for this method?

Reply:
Added.